### ISSUE TITLE
need a way to access the KG2 build code used for a specific build of KG2

### ISSUE THREAD

**Issue by saramsey** [RTXteam/RTX#1232]
It seems to me that it would be helpful to archive the KG2 build code as a part of the build process, like this:
```
tar cvzf /home/ubuntu/kg2-build/kg2-build-code.tar.gz -c /home/ubuntu/RTX/code kg2
aws s3 cp kg2-build-code-${KG2_RELEASE_VERSION}.tar.gz s3://rtx-kg2-versioned/
```

**Comment by saramsey** [comment]
This way, we always have the ability to reconstruct exactly what code was used in a given build of KG2, even if the code was locally modified or whatever.

**Comment by kvarforl** [comment]
I believe that github tags more or less do this! I forgot to push a tag for KG2.5.1 (since it was basically Kg2.5.0), but the previous releases should still be good examples.

If you navigate to the tags section, there are clickable `.zip` and `.tar.gz` icons that will download the contents of the whole repository from that point in time.

**Comment by saramsey** [comment]
OK, thanks, can you please update the post-build instructions in README.md to add a checklist item to tag the code used for the build?  This should be a documented step in the README.md.

**Comment by saramsey** [comment]
And yes, on reflection, if we can do this with GitHub tagging, that is the more mainstream and modern way of doing this.

**Comment by saramsey** [comment]
You are right, except that

1. The code and config files used for the build may not be exactly what is in github (if it was modified locally). This can happen inadvertantly if a local modification is left in or whatever.
2. Automating the git tagging requires git commit access to the RTX repo from within the build instance, which requires the setup script to do ssh key exchange. It can be done (I have done it with the Docker-based setup script for [kg2serv](https://github.com/RTXteam/kg2serv/blob/master/util/Dockerfile) but it is kind of a pain. 

I am open to counter-arguments. But it seems that in two lines of code (as shown above) we could tar up the source code and just save a copy on the versioned S3 bucket.

Nevertheless, I agree with you that we should *also* be tagging the code in GitHub, as a part of our standard operating procedure.

**Comment by kvarforl** [comment]
Makes sense to me! Just wanted to make sure we aren't duplicating effort, and your point about config files and lack of automatableness is very relevant. I think taring and saving to s3 is goo solution :)
